source/common/common/BUILD

```diff
@@ -133,10 +133,40 @@ envoy_cc_library(
     hdrs = ["logger.h"],
     external_deps = ["abseil_synchronization"],
     deps = [
+        ":base_logger_lib",
         ":lock_guard_lib",
         ":macros",
         ":non_copyable",
-    ],
+    ] + select({
+        "//bazel:android_logger": ["logger_impl_lib_android"],
+        "//conditions:default": ["logger_impl_lib_standard"],
+    }),
+)
+
+envoy_cc_library(
+    name = "base_logger_lib",
+    srcs = ["base_logger.cc"],
+    hdrs = ["base_logger.h"],
+)
+
+envoy_cc_library(
+    name = "logger_impl_lib_standard",
+    hdrs = ["standard/logger_impl.h"],
+    strip_include_prefix = "standard",
+)
+
+envoy_cc_library(
+    name = "logger_impl_lib_android",
+    srcs = select({
+        "//bazel:android_logger": ["android/logger_impl.cc"],
+        "//conditions:default": [],
+    }),
+    hdrs = select({
+        "//bazel:android_logger": ["android/logger_impl.h"],
+        "//conditions:default": [],
+    }),
+    strip_include_prefix = "android",
+    deps = [":base_logger_lib"],
```

Comment:
Given that the andoid sink in spdlog is `#ifdef __ANDROID__`, does it make sense to gate this logger on the equivalent of `//bazel::windows_x86_64` and `//bazel:apple` (instead of a flag that won't work in any other env) and following the pattern used for `envoy_cc_win32_library`? 

(The file watcher lib should get the same treatment for apple, but that's another PR.)

Reply:
That was my first idea, but @goaway pointed out that there might be cases where we might want the StandardLogger when `__ANDROID__` is defined, e.g running envoy as a process in an android device, rather than in a sandboxed application.

Reply:
Ok. Do we want to pull the selection logic out into helpers?

Reply:
I also debated about this. Given that it was only used once now and for the foreseeable future I opted for not introducing more helpers into the build system until they are needed. But if you think it is worth it, I am happy to add!

Reply:
I think the thing I'm getting it is whether having an empty envoy_cc_library (logger_impl_lib_android) when android logging is disabled causes some kind of problem for bazel (particular the query subcommand). It seemed like there was something related to that with win32 libs, but I'm a bit out of my depth when it comes to bazel. I suppose it can always be fixed later.

Reply:
I see. I was copying the pattern from win32 https://github.com/envoyproxy/envoy/blob/b78fc4e0edc696c2395b7eafbca8cbc62cb0f325/bazel/envoy_library.bzl#L151. Yeah, I agree, if there is a reported problem we can fix later.